scripts/run_sft.py

```diff
@@ -109,14 +109,17 @@ def main():
     torch_dtype = (
         model_args.torch_dtype if model_args.torch_dtype in ["auto", None] else getattr(torch, model_args.torch_dtype)
     )
+    params_offloaded = False
+    if accelerator.state.deepspeed_plugin and accelerator.state.deepspeed_plugin.offload_param_device != "none":
+        params_offloaded = True
 
     model_kwargs = dict(
         revision=model_args.model_revision,
         trust_remote_code=model_args.trust_remote_code,
         use_flash_attention_2=model_args.use_flash_attention_2,
         torch_dtype=torch_dtype,
         use_cache=False if training_args.gradient_checkpointing else True,
-        device_map=get_kbit_device_map(),
+        device_map=get_kbit_device_map() if not params_offloaded else None,
```

Comment:
Same suggestion here as above:
```suggestion
        device_map=get_kbit_device_map() if quantization_config is not None else None,
```